state/etcdstatedriver.go

```diff
@@ -176,16 +176,26 @@ func (d *EtcdStateDriver) ReadAll(baseKey string) ([][]byte, error) {
 	return [][]byte{}, err
 }
 
-func (d *EtcdStateDriver) channelEtcdEvents(watcher client.Watcher, rsps chan [2][]byte) {
+func (d *EtcdStateDriver) channelEtcdEvents(watcher client.Watcher, rsps chan [2][]byte, baseKey string) {
 	for {
 		// block on change notifications
 		etcdRsp, err := watcher.Next(context.Background())
 		if err != nil {
-			log.Errorf("Error %v during watch", err)
-			time.Sleep(time.Second)
+			log.Infof("Watch %s next failed: %v %v", baseKey, reflect.TypeOf(err), err)
+			switch err.(type) {
+			case *client.ClusterError:
+				// retry and wait for etcd cluster to recover!
+				time.Sleep(time.Second * 5)
```

Comment:
How did you choose this value for the sleep?

Reply:
The value for the sleep can be set any value.But this decides how many times the watcher will send request to the etcd until the cluster being recover.
The etcd cluster may be recovered any time after it broken.So i choose the value from my experiences.